app/models/exercise.rb

```diff
@@ -3,39 +3,77 @@ class Exercise < ApplicationRecord
   has_many :exercise_sets, dependent: :destroy
 
   def self.create_exercise(blueprint_id, workout_id, sets, user_id)
-    @blueprint = Blueprint.find(blueprint_id)
-    exercises = @blueprint.exercises.where(workout_id: nil)
+    blueprint = Blueprint.find(blueprint_id)
+    exercises = blueprint.exercises.where(workout_id: nil)
     exercises.each do |exercise|
       weight, reps =
-        exercise.set_weight_and_reps(user_id).values_at(:weight, :reps)
+        exercise.calculate_weight_and_reps(user_id).values_at(:weight, :reps)
+
       new_exercise =
         Exercise.create!(
           name: exercise["name"],
           instructions: exercise["instructions"],
           blueprint_id: exercise["blueprint_id"],
-          pic_url: exercise["pic_url"],
           workout_id: workout_id,
           weight: weight,
-          reps: reps
+          reps: reps,
+          user_id: user_id,
+          pic_url: exercise["pic_url"]
         )
       sets.times do
-        ExerciseSet.create_exercise_sets new_exercise["id"], user_id
+        ExerciseSet.create_exercise_sets new_exercise["id"],
+                                         user_id,
+                                         weight,
+                                         reps
       end
     end
   end
+  #The set_weight_and_reps function returns an object containing a weight and reps key. The values for these keys are assigned as follows
+  #If the user has never done this exercise before weight is automatically set to 5 and reps is automatically set to 8.
+  #If a user has done this exercise before then the comparison operator is utilized to keep track of how many sets were done below
+  #the exercise values, above the exercise values, or at the same values. If both @weight_success_rate and reps_success_rate = 0 then increase
+  #the reps by 1 unless reps >=10 then increase weight by 2.5 and reset reps to 8. If @weight_success_rate or reps_success_rate
+  #are > 0 then get the average of the previous exercise set group and set weight and reps to there respective averages. If
+  #@weight_success_rate or reps_success_rate < 0 but the user still completes at least half the sets at or above the exercise criteria
+  #weight and reps get set to the previous exercise weight and reps. If the user failed more than half the previous sets
+  #then weight and reps get set to the previous exercise sets averages.
+  def calculate_weight_and_reps(user_id)
```

Comment:
I think the description is worth keeping, but next time I'd work on making it shorter. It's just hard to digest that much information and really understand what's going on.

For the sake of brevity, it's okay to be vague and let the specific code in the function fill in any gaps you haven't explained in the comment.